core-bundle/src/Command/MigrateCommand.php

```diff
@@ -186,6 +196,30 @@ private function executeCommand(InputInterface $input): int
         return 0;
     }
 
+    private function hasWorkToDo(): bool
+    {
+        // There are some pending migrations
+        if ($this->migrations->hasPending()) {
```

Comment:
can't you use `iterator_count($this->migrations->getPending()) > 0`?

Reply:
Wouldn't it run through all migrations? I introduced `hasPending` to return early on the first pending one.

Reply:
You can use a `foreach` with an early return, this would also stop on the first migration that returns `true` for `shouldRun()`.

```suggestion
        foreach ($this->migrations->getPending() as $migration) {
```

Reply:
That's what `hasPending()` is doing, no?

Reply:
Yes, but with the foreach we don’t need an extra method for that.